docs/manual/advanced-features.tex

```diff
@@ -646,6 +646,9 @@
 warning that it cannot verify that the cast is correct.  The programmer may
 suppress the warning if the code is correct.
 
+One exception to this is if the annotation on the result type is polymorphic.  In this case, any
+explicit annotation between \<new> and a class name is treated as a constraint for the instantiation
```

Comment:
I do not understand what constraint this generates. Can you expand a bit, e.g. by expanding the example in the description with an illegal use?